src/vs/workbench/services/workingCopy/common/workingCopyFileService.ts

```diff
@@ -275,7 +279,18 @@ export class WorkingCopyFileService extends Disposable implements IWorkingCopyFi
 
 	//#region Path related
 
+	private readonly workingCopyProviders: WorkingCopyProvider[] = [];
+
+	registerWorkingCopyProvider(provider: WorkingCopyProvider): void {
+		this.workingCopyProviders.push(provider);
+	}
+
 	getDirty(resource: URI): IWorkingCopy[] {
+		const providerResources = flatten(this.workingCopyProviders.map(provider => provider(resource)));
+		if (providerResources.length) {
+			return providerResources;
```

Comment:
Why does this return early? We need to ask the `IWorkingCopyService` as well, no?